File: Easy/110. Balanced Binary Tree.js
var isBalanced = function(root) {
  let dfs = (node, level) => {
      if (!node) return level

      let left = dfs(node.left, level + 1)
      let right = dfs(node.right, level + 1)

      if (Math.abs(left - right) > 1) return Infinity

      return Math.max(left, right)
  }
  return dfs(root, 0) === Infinity ? false : true
};

File: Easy/110. Balanced Binary Tree.py
class Solution(object):
    def isBalanced(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if root is None:
            return True

        left = self.depth(root.left)
        right = self.depth(root.right)

        return abs(left - right) < 2 and self.isBalanced(root.left) and self.isBalanced(root.right)

    def depth(self, node):
        if node is None:
            return 0
        return max(self.depth(node.left), self.depth(node.right)) + 1